pkg/vat.go

```diff
@@ -162,16 +161,20 @@ func (v Vat) metrics() metricsReporter {
 }
 
 // match the protocol, ignoring namespace
-func matches(c Capability, proto string) bool {
+func matches(c Capability, proto protocol.ID) bool {
 	for _, p := range c.Protocols() {
-		if strings.HasSuffix(proto, string(p)) {
+		if hasSuffix(proto, p) {
 			return true
 		}
 	}
 
 	return false
 }
 
+func hasSuffix(s, suffix protocol.ID) bool {
+	return len(s) >= len(suffix) && s[len(s)-len(suffix):] == suffix
```

Comment:
This is a copy-paste of strings.HasSuffix that takes protocol.ID. I prefer this over converting protocol IDs to strings because it's less visual noise and doesn't perform a string-copy under the hood.